src/Nethermind/Nethermind.Clique/CliqueBridge.cs

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2018 Demerzel Solutions Limited
+ * This file is part of the Nethermind library.
+ *
+ * The Nethermind library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Nethermind library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+using System;
+using Nethermind.Blockchain;
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+
+namespace Nethermind.Clique
+{
+    public class CliqueBridge : ICliqueBridge
+    {
+        private const string CannotVoteOnNonValidatorMessage = "Cannot vote on non-validator node";
+        
+        private readonly CliqueBlockProducer _cliqueBlockProducer;
+        private readonly IBlockTree _blockTree;
+
+        public CliqueBridge(CliqueBlockProducer cliqueBlockProducer, IBlockTree blockTree)
+        {
+            if (_cliqueBlockProducer != null)
```

Comment:
Isn't it always false?

Reply:
in this particular situation we can have clique running but not mining and then block producer is null